controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -44,41 +44,56 @@ static std::string check_forbidden_chars(const std::string &str)
 }
 
 /*
-* @brief convert a mac address to hex notation 
+* @brief Convert a mac address to hex notation 
 * 
-* @param[in]  str String mac address
-* @return - none
+* @param[in] mac_string String mac address
+* @return Mac string in hex format
 */
-static void convert_mac_address_to_hex_notation(std::string &str)
+static void mac_to_hex_notation(std::string &mac_string)
 {
-    str.erase(std::remove(str.begin(), str.end(), ':'), str.end());
-    str = "0x" + str;
+    mac_string.erase(std::remove(mac_string.begin(), mac_string.end(), ':'), mac_string.end());
+    mac_string = "0x" + mac_string;
 }
 
 /*
-* @brief Checks that the given string represents a mac address 
+* @brief Convert a binary string to hex notation 
 * 
-* @param[in]  str String mac address
-* @return - true if string represents a mac address, else false.
+* @param[in] binary_string String in binary format
+* @return String in hex format
 */
-static bool validate_mac_address(const std::string &str)
+static void binary_to_hex(std::string &binary_string)
 {
-    int i                = 0;
-    int separators_count = 0;
-    for (auto &c : str) {
-        if (isxdigit(c)) {
-            i++;
-        } else if (c == ':') {
-            if (i == 0 || i / 2 - 1 != separators_count)
-                break;
-            ++separators_count;
-        } else {
-            separators_count = -1;
-        }
-    }
-    if (i == 12 && separators_count == 5)
-        return true;
-    return false;
+    binary_string.erase(0, 2);
+    std::string out;
+    for (uint i = 0; i < binary_string.size(); i += 4) {
+        int8_t n = 0;
+        for (uint j = i; j < i + 4; ++j) {
+            n <<= 1;
+            if (binary_string[j] == '1')
+                n |= 1;
+        }
+        if (n <= 9)
+            out.push_back('0' + n);
+        else
+            out.push_back('A' + n - 10);
+    }
+    binary_string = "0x" + out;
```

Comment:
I liked your implementation,
but how can you do an assignment of const reference?

Reply:
I think Moran meant to `return output` instead of setting `binary_string = output`.

Reply:
Actually, it's better if all the conversion functions have a similar signature. So if this function is changed to return the output rather than taking an in-out argument, it's better to do that everywhere. Or if you prefer not to do that, it's better to keep it as an in-out argument here as well.